src/query/functions/temporal/base_test.go

```diff
@@ -351,6 +353,158 @@ func setup(numBlocks int, duration time.Duration, nextBound int) *testContext {
 	}
 }
 
+// TestBaseWithDownstreamError checks that we handle errors from blocks correctly
+func TestBaseWithDownstreamError(t *testing.T) {
```

Comment:
> but this would be a good test to run on other functions as well

Probably, but I'd rather fix the panic first :). I can put up a separate issue for it.

Reply:
Sounds good 👍